garrysmod/lua/includes/extensions/net.lua

```diff
@@ -234,3 +234,20 @@ function net.ReadType( typeid )
 	error( "net.ReadType: Couldn't read type " .. typeid )
 
 end
+
+do
+	local mask = 0x100000000
+
+	function net.WriteSteamID64( steamid64 )
+		local unique_part = tonumber( string.sub(steamid64, 6) )
+		local high = unique_part % mask
+		local low = math.floor(unique_part / mask)
+
+		net.WriteUInt(high, 32) net.WriteUInt(low, 8)
+	end
+
+	function net.ReadSteamID64()
+		local high, low = net.ReadUInt(32), net.ReadUInt(8)
+		return "76561" .. tostring(low * mask + high)
```

Comment:
Perhaps on the Write side you can check if the SteamID64 _doesn't_ start with 76561 and if so, `net.WriteBool(true)` then write the missing 24 bits Rubat mentioned. On the read side, you read that bool, if it's true you read the missing 24 bits, otherwise you assume `76561`.